uberfire-extensions/uberfire-metadata/uberfire-metadata-backends/uberfire-metadata-backend-infinispan/src/main/java/org/uberfire/ext/metadata/backend/infinispan/proto/KObjectMarshaller.java

```diff
@@ -0,0 +1,287 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.uberfire.ext.metadata.backend.infinispan.proto;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.infinispan.protostream.MessageMarshaller;
+import org.infinispan.protostream.descriptors.Descriptor;
+import org.infinispan.protostream.descriptors.FieldDescriptor;
+import org.infinispan.protostream.descriptors.JavaType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.ext.metadata.model.KObject;
+import org.uberfire.ext.metadata.model.KProperty;
+import org.uberfire.ext.metadata.model.impl.KObjectImpl;
+import org.uberfire.ext.metadata.model.impl.KPropertyImpl;
+import org.uberfire.ext.metadata.model.schema.MetaObject;
+import org.uberfire.java.nio.base.version.VersionHistory;
+import org.uberfire.java.nio.file.attribute.FileTime;
+
+import static java.util.stream.Collectors.joining;
+import static java.util.stream.Collectors.toList;
+import static org.uberfire.ext.metadata.backend.infinispan.utils.AttributesUtil.toKPropertyFormat;
+import static org.uberfire.ext.metadata.backend.infinispan.utils.AttributesUtil.toProtobufFormat;
+
+public class KObjectMarshaller implements MessageMarshaller<KObject> {
+
+    private Logger logger = LoggerFactory.getLogger(KObjectMarshaller.class);
+
+    public static final String CLUSTER_ID = toProtobufFormat(MetaObject.META_OBJECT_CLUSTER_ID);
+    public static final String SEGMENT_ID = toProtobufFormat(MetaObject.META_OBJECT_SEGMENT_ID);
+
+    public static final String CHECKIN_COMMENT = "checkinComment";
```

Comment:
Is there a need to make these public? I didn't find any usages, so perhaps these should be all private.